### PR TITLE
[SourceKit/CodeCompletion] Add an option to emit annotated description

### PR DESCRIPTION
**Pull request by rintaro**
rdar://problem/60801189

### CODE REVIEW

**Comment by akyrtzi** [comment]
Some general thoughts:

* For primary correctness testing we should use `swift-ide-test` tests, along with an explicit option to enable annotations. We should add a `sourcekitd-test` only for sanity checking that the annotations are enabled via its interface.
* There be should a test to ensure that we get the annotations cached for symbols from module imports.
* Possibly shorten `callarg.argument_label` -> `callarg.label` and `callarg.parameter_name` -> `callarg.param`
* Mark type identifiers separately (so that they are marked in closure types) and indicate if they are system or not. Strawman:"<callarg.type>(<typeid.sys>Int</typeid.sys>) -&gt; <typeid&gt;MyFoo</typeid&gt;</callarg.type></callarg>)"
* If `throws`/`rethrows` shows up, it should be marked as keyword.

File: test/IDE/complete_annotation.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+func testMember(value: MyStruct) {
+  value.#^EXPR_MEMBER^#
+}
+// EXPR_MEMBER: Begin completions, 7 items
+// EXPR_MEMBER-DAG: Keyword[self]/CurrNominal:          <keyword>self</keyword>; name=self
+// EXPR_MEMBER-DAG: Decl[InstanceVar]/CurrNominal:      <name>propNormal</name>; name=propNormal
+// EXPR_MEMBER-DAG: Decl[InstanceVar]/CurrNominal:      <name>propFunction</name>; name=propFunction
+// EXPR_MEMBER-DAG: Decl[InstanceMethod]/CurrNominal: <name>labelNameParamName</name>(<callarg><callarg.label>label</callarg.label> <callarg.param>param</callarg.param>: <callarg.type>(<keyword>inout</keyword> <typeid.sys>Int</typeid.sys>) <keyword>throws</keyword> -&gt; <typeid.user>MyStruct</typeid.user></callarg.type></callarg>)<keyword> rethrows</keyword>; name=labelNameParamName(label: (inout Int) throws -> MyStruct) rethrows
+// EXPR_MEMBER-DAG: Decl[InstanceMethod]/CurrNominal: <name>labelName</name>(<callarg><callarg.label>label</callarg.label>: <callarg.type>(<attribute>@autoclosure </attribute>() -&gt; <typeid.sys>Int</typeid.sys>) -&gt; <typeid.sys>Int</typeid.sys></callarg.type></callarg>); name=labelName(label: (@autoclosure () -> Int) -> Int)
```
Comment (akyrtzi) [inline test/IDE/complete_annotation.swift:72]: There is a space inside the attribute wrap, "<attribute\>@autoclosure </attribute\>", ideally the tag would not include the space.

**Review by akyrtzi — APPROVED** [review]
A minor issue with the attribute wrap, and make sure to have test coverage that annotation is coming through the completion cache, otherwise LGTM! 👍

File: test/IDE/complete_annotation.swift (hunk 1 above)
Comment (rintaro) [reply test/IDE/complete_annotation.swift:72]: Same for `<keyword> rethrows</keyword>`. I will see if fixing this affects the performance. But to clarify, even if we don't fix this, it should not affect the clients practically, right?

Comment (akyrtzi) [reply test/IDE/complete_annotation.swift:72]: I can't say for sure either way.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: test/IDE/complete_annotation.swift (hunk 1 above)
Comment (rintaro) [reply test/IDE/complete_annotation.swift]: Done https://github.com/apple/swift/blob/339ebbfa89a3648220aeb765be0c19315d36f140/lib/IDE/CodeCompletionResultPrinter.cpp#L70-L84

**Comment by compnerd** [comment]
@rintaro this seems to have regressed the Windows build: https://ci-external.swift.org/job/oss-swift-windows-x86_64-vs2019/739/

**Comment by compnerd** [comment]
Please run full tests in the future - the smoke test is not reliable.

**Comment by rintaro** [comment]
@compnerd Thanks for notifying. Looking.

**Comment by rintaro** [comment]
For now, disabling the test case in Windows. https://github.com/apple/swift/pull/30855
@compnerd This looks like newline character issue. Do you have any insight about this?

**Comment by rintaro** [comment]
Testing with `--strip-trailing-cr` https://github.com/apple/swift/pull/30858

**Pull request merged**